src/kafka/inflight_activation_batcher.rs

```diff
@@ -50,13 +50,22 @@ impl Reducer for InflightActivationBatcher {
     async fn reduce(&mut self, t: Self::Input) -> Result<(), anyhow::Error> {
         let runtime_config = self.runtime_config_manager.read().await;
         let task_name = &t.activation.taskname;
-        if !runtime_config.drop_task_killswitch.contains(task_name) {
-            self.buffer.push(t);
-        } else {
+
+        if runtime_config.drop_task_killswitch.contains(task_name) {
             metrics::counter!("filter.drop_task_killswitch", "taskname" => task_name.clone())
                 .increment(1);
+            return Ok(());
         }
 
+        if let Some(expires_at) = t.expires_at {
+            if Utc::now() > expires_at {
+                metrics::counter!("upkeep.expired").increment(1);
```

Comment:
I think we should call this `upkeep.expired_early` or something like that? So we can tell the difference?

Reply:
Renamed to `filter.expired_at_consumer` 🤔